tests/cobs_test.cc

```diff
@@ -0,0 +1,145 @@
+#include "../inc/cobs.h"
+
+/*
+Throw exceptions so we can catch them with gdb (catch catch).
+*/
+void throw_assert(bool condition, const char * const remark) {
+  if (condition == false) throw remark;
+}
+
+/*
+Fill a byte buffer with a given value.
+*/
+void fill(uint8_t * const b, size_t l, uint8_t v) {
+  for (size_t i = 0; i < l; i++) {
+    b[i] = v;
+  }
+}
+
+/*
+Check that a byte buffer contains only a given value.
+*/
+void cmpb(uint8_t * const b, size_t l, uint8_t v) {
+  for (size_t i = 0; i < l; i++) {
+    throw_assert(b[i] == v, "");
+  }
+}
+
+int main() {
+  const uint8_t X = 0xdd; // non-zero payload
+  const uint8_t F = 0xcc; // fill
+
+  // Encode the empty packet.
+  {
+    uint8_t b1[1000]; fill(b1, sizeof(b1), X);
```

Comment:
In general, I place each statement on a separate line so that is easier to diff (sometimes) but mostly for stepping through a debugger. Note that these are two separate instructions: an initialization followed by a set.

I keep with that guideline so what I would do is:
```cpp
uint8_t b1[1000];
fill(b1, sizeof(b1), X);
```
even though this is a simple case in order to be consistent.
I'd also use std::array (or std::vector) instead of std::list to allow fast random access.